libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -2222,6 +2224,21 @@ def write_hwdef_header(outfilename):
                 if fnmatch.fnmatch(d, r):
                     error("Missing required DMA for %s" % d)
 
+    f.close()
+    # see if we ended up with the same file, on an unnecessary reconfigure
+    try:
+        if filecmp.cmp(outfilename, tmpfile):
+            print("No change in hwdef.h")
+            os.unlink(tmpfile)
+            return
+    except FileNotFoundError:
+        pass
+    try:
+        os.unlink(outfilename)
+    except FileNotFoundError:
+        pass
```

Comment:
```suggestion
```

`os.rename` quite happily blats the second file.

Reply:
you tested on windows? on cygwin? on macos?
better to handle the error than just guess